app/components/createContainer/createContainerController.js

```diff
@@ -284,19 +446,23 @@ function ($q, $scope, $state, $stateParams, $filter, Container, ContainerHelper,
   }
 
   $scope.create = function () {
-    $('#createContainerSpinner').show();
+    confirmCreateContainer(function (doIt) {
```

Comment:
I'd rather see this part using a promise than a callback as this part is quite hard to read.

Like:

```javascript
    confirmCreateContainer()
    .then(function success(confirm) {
      if (!confirm) {
        return false;
      }

      $('#createContainerSpinner').show();
      var accessControlData = $scope.formValues.AccessControlData;
      var userDetails = Authentication.getUserDetails();
      var isAdmin = userDetails.role === 1 ? true : false;

      if (!validateForm(accessControlData, isAdmin)) {
        $('#createContainerSpinner').hide();
        return;
      }

      var config = prepareConfiguration();
      createContainer(config, accessControlData);
    })
    .catch(function error(err) {
      Notifications.error('Failure', err, 'Unable to create container');
    })
    .finally(function final() {
      $('#createContainerSpinner').hide();
    });
```

You'd need to rewrite the `confirmCreateContainer()` function to use `$q`.

```javascript
  function confirmCreateContainer() {
    var deferred = $q.defer();

    Container.query({ all: 1, filters: { name: [$scope.config.name] } }).$promise
    .then(function success(data) {
      // The section where we check if another container with that name exists
      // should be externalized as another function
      // Should have:
      // var existingContainer = getContainerWithNameInContainers($scope.config.name)
      // if (existingContainer) {
      //  ModalService.confirmDeletion(...)
      // }
      var confirmDialog = false;
      var container;

      for (var c in data) {
        for (var n in data[c].Names) {
          if (data[c].Names[n] === '/' + $scope.config.name) {
            confirmDialog = true;
            container = data[c];
          }
        }
      }
      if (confirmDialog) {
        // I've already updated the text of the modal but not the button.
        // Note that the confirm button text should be 'Replace' instead of 'Remove'.
        ModalService.confirmDeletion(
          'A container with the same name already exists. Portainer can automatically remove it and re-create one. Do you want to replace it?',
          function onConfirm(confirmed) {
            if(!confirmed) { deferred.resolve(false); }
            else {
              // Remove old container
              ContainerService.remove(container, true)
              .then(function success(data) {
                Notifications.success('Container Removed', container.Id);
                deferred.resolve(true);
              })
              .catch(function error(err) {
                deferred.reject({ msg: 'Unable to remove container', err: err });
              });
            }
          }
        );
      } else {
        deferred.resolve(true);
      }
    })
    .catch(function error(err) {
      deferred.reject({ msg: 'Unable to retrieve containers', err: err });
    });

    return deferred.promise;
  }
```